### PR TITLE
Fix Options Clone

### PR DESCRIPTION
**Pull request by tylertreat**
Ensure Options.Clone() only initializes Users and Routes when the
Options it's cloning has them initialized.

@kozlovic

### CODE REVIEW

**Comment by derekcollison** [comment]
LGTM

File: server/auth_test.go [hunk 1]
```diff
@@ -49,6 +49,23 @@ func TestUserClone(t *testing.T) {
...
+	}
+
+	clone := user.clone()
+
+	if clone.Permissions.Publish != nil {
+		t.Fatalf("Expected Publish to be nil, got: %v", clone.Permissions.Publish)
+	}
+	if clone.Permissions.Subscribe != nil {
+		t.Fatalf("Expected Subscribe to be nil, got: %v", clone.Permissions.Subscribe)
+	}
```
Comment (kozlovic) [inline server/auth_test.go]: Can you add a test where Permissions is nil (not empty like in this test) and that the clone's Permissions is nil too?
